Shared spack (#47615)
spack/spack: lib/spack/docs/chain.rst

.. code-block:: yaml

  upstreams:
    install-tree-1:
      install_tree: /path/to/other/dir/containing/.spack-db
    install-tree-2:

>       install_tree: /path/to/another/dir/containing/.spack-db

Reviewer: ```suggestion
    install-tree-1:
      install_tree: /path/to/other/dir/containing/install/tree
    install-tree-2:
      install_tree: /path/to/another/dir/containing/install/tree
```

> The ``install_tree`` must point to a directory containing ``.spack-db``, for example ``install_tree`` defined in :ref:`config.yaml <config-yaml>`.

Reviewer: ```suggestion
The ``install_tree`` must point to a directory containing a spack install tree, as defined in :ref:`config.yaml <config-yaml>`.

.. note:
   
   The ``install_tree`` directory is the directory containing the ``.spack-db`` hidden directory. This can be used to confirm you have the correct directory level listed for your upstream.
```

I think we should define this in terms of the configured directory, not in terms of the hidden directory that they shouldn't ever need to think about. I've retained the reference as a note to help them confirm.

Author: Upcoming commits will remove references to `.spack-db`, but I think there has to be a precise way to tell users what their install dir is, so I added `spack location --install-root` (the doc then says to do this or `spack config get config`, as before this PR that wouldn't work).

(I collapsed the other comment since I think we can track this in one convo)

.. code-block:: yaml

  upstreams:
    install-tree-1:
      install_tree: /path/to/another/spack/install/tree
    install-tree-2:

>       install_tree: /path/to/old/spack/opt/spack

Reviewer: ```suggestion
      install_tree: /path/to/yet/another/spack/install/tree
```
I don't want to preserve references to the old format where we don't need them, since hopefully eventually it will be long-forgotten.
